distro/wildfly/subsystem/src/main/java/org/camunda/bpm/container/impl/jboss/service/MscExecutorService.java

```diff
@@ -30,79 +32,88 @@
 import org.jboss.msc.service.StartContext;
 import org.jboss.msc.service.StartException;
 import org.jboss.msc.service.StopContext;
-import org.jboss.msc.value.InjectedValue;
-import org.jboss.threads.ExecutionTimedOutException;
 
 
 public class MscExecutorService implements Service<MscExecutorService>, ExecutorService {
 
   private static Logger log = Logger.getLogger(MscExecutorService.class.getName());
-  
-  private final InjectedValue<ManagedQueueExecutorService> managedQueueInjector = new InjectedValue<ManagedQueueExecutorService>();
-  
+
+  protected final Supplier<ManagedQueueExecutorService> managedQueueSupplier;
+  protected final Consumer<ExecutorService> provider;
+
   private long lastWarningLogged = System.currentTimeMillis();
 
+  public MscExecutorService(Supplier<ManagedQueueExecutorService> managedQueueSupplier, Consumer<ExecutorService> provider) {
+    this.managedQueueSupplier = managedQueueSupplier;
+    this.provider = provider;
+  }
+
+  @Override
   public MscExecutorService getValue() throws IllegalStateException, IllegalArgumentException {
     return this;
   }
 
+  @Override
   public void start(StartContext context) throws StartException {
     // nothing to do
+    provider.accept(this);
   }
 
+  @Override
   public void stop(StopContext context) {
-    // nothing to do    
+    // nothing to do
+    provider.accept(null);
```

Comment:
🙃 I guess now there is something to do.

```suggestion
    provider.accept(null);
```